lib/src/core/engine.dart

```diff
@@ -285,29 +289,37 @@ class Engine extends Disposable with EventsEmittable<EngineEvent> {
         ));
       }
 
-      final iceConnected =
-          primary?.pc.iceConnectionState?.isConnected() ?? false;
+      final iceConnected = primary?.pc.connectionState?.isConnected() ?? false;
 
       logger.fine('Reconnect: iceConnected: $iceConnected');
 
       if (!iceConnected) {
         logger.fine('Reconnect: Waiting for primary to connect...');
 
-        await events.waitFor<EngineIceStateUpdatedEvent>(
-          filter: (event) => event.isPrimary && event.iceState.isConnected(),
+        await events.waitFor<EnginePeerStateUpdatedEvent>(
+          filter: (event) => event.isPrimary && event.state.isConnected(),
           duration: Timeouts.iceRestart,
+          onTimeout: () => throw ConnectException(),
         );
       }
+    }
 
-      logger.fine('Reconnect: success');
-      _connectionState = ConnectionState.connected;
-      events.emit(const EngineReconnectedEvent());
-      _reconnectAttempts = 0;
+    try {
+      _updateConnectionState(ConnectionState.reconnecting);
+      await Utils.retry<void>(
+        (tries, errors) {
+          logger.fine('Retrying connect sequence remaining ${tries} tries...');
+          return sequence();
+        },
+        retryCondition: (_, __) =>
+            _connectionState == ConnectionState.reconnecting,
+        tries: 10,
```

Comment:
I think this is what we discussed in the meeting about not making too many retries before a full reconnect. Making a note when I remember it :-)